internal/cmd/eventgen/schemadefs.go

```diff
@@ -0,0 +1,210 @@
+// Copyright Axis Communications AB.
+//
+// For a full list of individual contributors, please see the commit history.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	_ "embed"
+	"fmt"
+	"io"
+	"text/template"
+
+	"github.com/Masterminds/semver"
+	jsschema "github.com/lestrrat-go/jsschema"
+
+	"github.com/eiffel-community/eiffelevents-sdk-go"
+	"github.com/eiffel-community/eiffelevents-sdk-go/internal/codetemplate"
+)
+
+// schemaDefinitionRenderer renders a schema definition to a Go source file.
+type schemaDefinitionRenderer interface {
+	EventType() string
+	Filename() string
+	Render(schema io.Reader, outputFile string) error
+	Version() *semver.Version
+}
+
+// definitionFile holds common fields and methods to support rendering
+// any kind of schema definition to a Go source file. The struct can
+// be embedded in a new struct that implmements the Render method from
+// schemaDefinitionRenderer to get a working renderer.
+type definitionFile struct {
+	filename  string
+	eventType string
+	version   *semver.Version
+}
+
+func (edf *definitionFile) EventType() string {
+	return edf.eventType
+}
+
+func (edf *definitionFile) Filename() string {
+	return edf.filename
+}
+
+func (edf *definitionFile) Version() *semver.Version {
```

Comment:
In the two functions above you return a copy of the value, but here you return the pointer. 
Do you want to do this, or is it irrelevant?

Reply:
It's a conscious decision mainly based on semver.NewVersion and semver.MustParse both returning *semver.Version (I figured following the pattern of the constructor functions would result in as little trouble as possible). But yes, this is at least a theoretical problem. While most semver.Version methods that perform mutations return a new value, UnmarshalJSON notably doesn't. I don't consider this a problem but I can look into it if you want.